.ci/set-milestone-on-referenced-issue.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+set -e
+
+source ./.ci/util.sh
+
+if [[ -z $1 ]]; then
+  echo "[ERROR] Pull request number is not set."
+  echo "Usage: $BASH_SOURCE <pull request number>"
+  exit 1
+fi
+
+PULL_REQUEST_NUMBER=$1
+echo "PULL_REQUEST_NUMBER=$PULL_REQUEST_NUMBER"
+
+checkForVariable "GITHUB_TOKEN"
+
+echo "Fetching all commit messages from pull request #$PULL_REQUEST_NUMBER."
+PULL_REQUEST_COMMITS=$(curl -s \
+  -H "Accept: application/vnd.github+json" \
+  -H "Authorization: Bearer $GITHUB_TOKEN"\
+  https://api.github.com/repos/checkstyle/checkstyle/pulls/"$PULL_REQUEST_NUMBER"/commits)
+PULL_REQUEST_COMMIT_MESSAGES=$(echo "$PULL_REQUEST_COMMITS" | jq -r .[].commit.message)
+echo "PULL_REQUEST_COMMIT_MESSAGES=
+$PULL_REQUEST_COMMIT_MESSAGES"
+
+echo "Extracting all issue numbers from commit messages."
+ISSUE_NUMBERS=$(echo "$PULL_REQUEST_COMMIT_MESSAGES" | grep -oE "#[0-9]+" | cut -c2-)
```

Comment:
e5b56f8dd158350df839198fd75d5eb40935011e
Since we aren't checking for `Issue #\d` or `Pull #\d`, won't the example commit above be incorrectly picked up. Was there a reason for not finding `Issue` or `Pull`?

Reply:
Oof, at beginning of PR we used to check for those and then removed them and I completely forgot to start checking in script for that:
https://github.com/checkstyle/checkstyle/blob/d46afc92ef6270f12cf00a8a9370d036952b5d5b/.github/workflows/set-milestone.yml#L22-L26

I will add a check.

Reply:
Done.

PR: https://github.com/stoyanK7/checkstyle/pull/56
Actions job: https://github.com/stoyanK7/checkstyle/actions/runs/4315749586/jobs/7530566547#step:3:8
```
Run ./.ci/set-milestone-on-referenced-issue.sh 56
PULL_REQUEST_NUMBER=56
Fetching commit messages that match '^(Pull|Issue) #[0-9]+' from PR #56.
PULL_REQUEST_COMMIT_MESSAGES=
Pull #56: heello
Extracting issue numbers from commit messages.
ISSUE_NUMBERS=
56
Fetching latest milestone.
MILESTONE_NUMBER=11
MILESTONE_TITLE=10.8.1
Setting milestone 10.8.1 to issue #56.
```